### DISCUSSION TITLE
How to save listview items in txt file and load it in listview again if we want in WPF?

### DISCUSSION THREAD

**Discussion by VahidEra** [dotnet/wpf#7453] [Q&A] [answered]
Hi guys,
I have a list view in my WPF project that I want to save its items in a text file that if I needed when I wanted to show it in a list view.
My ListView has two columns, called NAME and Address, Which by an Open File Dialog indicate the name and the selected file address in the listview
```
             Public Class MyRowData
                  Public Property name As String
                  Public Property address As String
             End Class

             Private Sub itmAddToPlayList_Click(sender As Object, e As RoutedEventArgs) Handles itmAddToPlayList.Click
                  ListView1.Items.Add(New MyRowData With {.name = openFileDialog1.SafeFileName, .address = openFileDialog1.FileName})
             End Sub
```

I am easily accomplished in Windows Form by the following code:

```
             If SaveFileDialog2.ShowDialog = DialogResult.OK Then
                    Dim myWriter As New IO.StreamWriter(SaveFileDialog2.FileName)
                    For Each myItem As ListViewItem In ListView1.Items
                        myWriter.WriteLine(myItem.Text & "#" & myItem.SubItems(1).Text)
                    Next
                    myWriter.Close()        
              End If
```

And also use this code to load the text file to my listview:

```
               If OpenFileDialog3.ShowDialog = DialogResult.OK Then                             
                    Dim myCoolFileLines() As String = IO.File.ReadAllLines(OpenFileDialog3.FileName)
                     For Each line As String In myCoolFileLines
                             Dim lineArray() As String = line.Split("#")
                             Dim newItem As New ListViewItem(lineArray(0))
                             newItem.SubItems.Add(lineArray(1))
                             ListView1.Items.Add(newItem)
                     Next
               End If
```

>**But as I'm new to wpf I have no idea how to do this thanks guys for your help.**

**Comment by miloush** [comment, marked as answer]
Like in your previous question, you have to use the type you created. For saving you do `For Each myItem as MyRowData in ListView1.Items` and then you can write out  `myItem.name` and `myItem.address`. And for loading you add `New MyRowData with {.name = lineArray(0), .address = lineArray(1)}`

**Reply by VahidEra** [reply]
@miloush Dear miloush,
you are the best, the best never rest.
This is exactly what I wanted And as always, you provided a simple and effective solution.
This is the second time you help me and solve my problem, because I'm in the new WPF, I ask a lot of questions. My dear friend also has another problem and I'm sure, as always, you know it. Hope to help me. I will come up soon and mention you. Thank you for all your help.
